src/domain/authentication/index.types.d.ts

```diff
@@ -0,0 +1,37 @@
+type SessionId = string & { readonly brand: unique symbol }
+type SchemaId = string & { readonly brand: unique symbol }
+
+type AuthenticationError = import("./errors").AuthenticationError
+
+type IdentityPassword = string & { readonly brand: unique symbol }
+
+type KratosUserId = string & { readonly brand: unique symbol }
+type SessionToken = string & { readonly brand: unique symbol }
+
+type Identity = {
+  id: KratosUserId
+  schemaId: SchemaId
```

Comment:
Why is `schemaId` required in the app use cases? Isn't this an internal detail?


Do we have logic that switches on this field? What could a domain based representation look like? Perhaps 2 types `EmailIdentity` and `PasswordIdentity`? Or if they must be unified then perhaps `IdentityType` that is an explicitly declared enum.

Reply:
yup good point